pkg/cli/server/root.go

```diff
@@ -861,6 +861,37 @@ func validateGC(config *config.Config, log zlog.Logger) error {
 
 			return zerr.ErrBadConfig
 		}
+
+		if err := validateGCRules(subPath.Retention, log); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func validateGCRules(gc config.ImageRetention, log zlog.Logger) error {
+	for _, policy := range gc.Policies {
+		for _, pattern := range policy.RepoNames {
+			if ok := glob.ValidatePattern(pattern); !ok {
+				log.Error().Err(glob.ErrBadPattern).Str("pattern", pattern).
+					Msg("retention repo glob pattern could not be compiled")
+
+				return zerr.ErrBadConfig
+			}
+		}
+
+		for _, tagRule := range policy.TagsRetention {
+			for _, regex := range tagRule.Names {
+				_, err := regexp.Compile(regex)
```

Comment:
Performance-wise, would it pay off to actually cache the compiled regexp for each rule and re-use it later instead of re-compiling it?

Reply:
yes, that would be nice.